modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java

```diff
@@ -4611,6 +4611,24 @@ else if (log.isInfoEnabled())
             log.info("\"" + registry + "\" not found.");
     }
 
+    /** {@inheritDoc} */
+    @Override public void configureHitRateMetric(String name, long rateTimeInterval) {
```

Comment:
I believe that we shouldn't configure HitRate metrics at all because there is no valid cases for it. See corresponding discussion on dev-list. So only histograms should be configurable.

Reply:
I think we should keep the ability to configure hitrate metric.
Let's take a look at the `WalLoggingRate`.
The user may want to change that interval to measure the rate more clear or for a longer period of time: 

```
        walLoggingRate = mreg.hitRateMetric("WalLoggingRate",
            "Average number of WAL records per second written during the last time interval.",
            rateTimeInterval,
            subInts);
```

Reply:
Proper way to measure the rate for a longer period time is using external tools that can aggregate metric values for any period. It is very simple arithmetic operation. So there is no need in additional functionality in Ignite. Moreover, changing of time interval could lead to complication of calculation of valid hit rate value for long period of time.

Reply:
We have this configuration for now.
And it's not deprecated or something.
All I do - implementing a universal approach for hitrate configuration.

I think we should keep this.
It doesn't complicate final solution in any way.

```
    /**
     * Sets time interval for {@link DataRegionMetrics#getAllocationRate()}
     * and {@link DataRegionMetrics#getEvictionRate()} monitoring purposes.
     * <p>
     * For instance, after setting the interval to 60 seconds,
     * subsequent calls to {@link DataRegionMetrics#getAllocationRate()}
     * will return average allocation rate (pages per second) for the last minute.
     *
     * @param metricsRateTimeInterval Time interval used for allocation and eviction rates calculations.
     * @return {@code this} for chaining.
     */
    public DataRegionConfiguration setMetricsRateTimeInterval(long metricsRateTimeInterval) {
        this.metricsRateTimeInterval = metricsRateTimeInterval;

        return this;
    }
```